src/test/scala/chiselTests/ResetSpec.scala

```diff
@@ -0,0 +1,50 @@
+// See LICENSE for license details.
+
+package chiselTests
+
+import chisel3._
+import chisel3.experimental.{IO, RawModule}
+import chisel3.util.{Counter, Queue}
+import chisel3.testers.BasicTester
+
+class ResetAgnosticModule extends RawModule {
+  val clk = IO(Input(Clock()))
+  val rst = IO(Input(Reset()))
+  val out = IO(Output(UInt(8.W)))
+
+  val reg = withClockAndReset(clk, rst)(RegInit(0.U(8.W)))
+  reg := reg + 1.U
+  out := reg
+}
+
+
+class ResetSpec extends ChiselFlatSpec {
+
+  behavior of "Reset"
+
+  it should "allow writing modules that are reset agnostic" in {
```

Comment:
Is it also worth testing that this work with the implicit clock and reset in Module?

Reply:
All tests using a `Module` are testing this, in particular, `AsyncResetQueueTester` ensures that `Queue` works with async reset.